src/test/kotlin/com/atlassian/performance/tools/virtualusers/LoadTestIT.kt

```diff
@@ -96,34 +96,32 @@ class LoadTestIT {
     private fun <T> testWithJira(
         test: (URI) -> T
     ): T {
-        val docker = DockerClientBuilder.getInstance().build()
-        return docker
-            .createNetworkCmd()
-            .withName(UUID.randomUUID().toString())
-            .execAsResource(docker)
-            .use { network ->
-                val networkId = network.response.id
-                val dbImage = SudoSshUbuntuImage(docker, networkId, listOf(3306))
-                val jiraImage = SudoSshUbuntuImage(docker, networkId, listOf(8080))
-                dbImage.runInUbuntu { db ->
-                    jiraImage.runInUbuntu { jira ->
-                        test(runJiraServer(jira, db))
-                    }
-                }
+        fun exposePort(
+            port: Int
+        ) = Consumer { container: GenericContainer<*> ->
+            container.addExposedPort(port)
+        }
+
+        val dbContainer = SshUbuntuContainer(exposePort(3306))
+        val jiraContainer = SshUbuntuContainer(exposePort(8080))
+        return dbContainer.start().use { db ->
```

Comment:
Just an idea. We could avoid the nested containers by introducing a container that would allow registering all the docker containers:

```
val env = Environment()
env.register(dbContainer)
env.register(jiraContainer)
// env takes care to start and close containers in the right order
env.use{

}
```
It would be an overkill for two containers but it would improve readability for 3+ containers.